### PR TITLE
OP-159 externalize jwt secrets only

### PR DESCRIPTION
**Pull request by mwithi**
Preparing for release 1.12.0... see OP-159, paired with https://github.com/informatici/openhospital-core/pull/862.

- [x] Get rid of hardcoded secrets.

### CODE REVIEW

File: src/main/java/org/isf/security/jwt/TokenProvider.java [hunk 1]
```diff
@@ -50,6 +52,9 @@ public class TokenProvider {
 
     private final Logger log = LoggerFactory.getLogger(TokenProvider.class);
 
+	@Autowired
+	private Environment env;
```
Comment (mwithi) [inline src/main/java/org/isf/security/jwt/TokenProvider.java:56]:
```suggestion
    @Autowired
    private Environment env;
```

File: src/main/java/org/isf/security/jwt/JWTAuthenticationFilter.java [hunk 2]
```diff
@@ -42,11 +44,15 @@
...
+	@Autowired
+	private Environment env;
 
     private AuthenticationManager authenticationManager;
 
-    private long EXPIRATION_TIME=SecurityConstants.EXPIRATION_TIME;
-    private String SECRET=SecurityConstants.SECRET;
+    private long EXPIRATION_TIME = SecurityConstants.EXPIRATION_TIME;
+    
+
```
Comment (mwithi) [inline src/main/java/org/isf/security/jwt/JWTAuthenticationFilter.java:55]:
```suggestion
    private long EXPIRATION_TIME = SecurityConstants.EXPIRATION_TIME;
    
```

File: src/main/java/org/isf/security/jwt/JWTAuthenticationFilter.java [hunk 3]
```diff
@@ -42,11 +44,15 @@
 import io.jsonwebtoken.SignatureAlgorithm;
 
 public class JWTAuthenticationFilter extends UsernamePasswordAuthenticationFilter {
+	
+	@Autowired
+	private Environment env;
```
Comment (mwithi) [inline src/main/java/org/isf/security/jwt/JWTAuthenticationFilter.java:49]:
```suggestion
    @Autowired
    private Environment env;
```

File: README.md [hunk 4]
```diff
@@ -20,11 +20,12 @@ For the moment, to build this project you should
         cd openhospital-api
         mvn clean install -DskipTests=true
         
- 3. prepare settings from rsc/*.dist files
+ 3. prepare settings from each rsc/*.dist files
```
Comment (dbmalkovsky) [inline README.md:23]:
```suggestion
 3. prepare settings from each rsc/*.dist file
```

File: src/main/java/org/isf/security/jwt/JWTAuthorizationFilter.java [hunk 5]
```diff
@@ -42,10 +43,13 @@ public class JWTAuthorizationFilter extends BasicAuthenticationFilter {
...
 	private String HEADER_STRING = SecurityConstants.HEADER_STRING;
 	private String TOKEN_PREFIX = SecurityConstants.TOKEN_PREFIX;
-	private String SECRET = SecurityConstants.SECRET;
 	
+	@Autowired
+	private Environment env;
+
 	@Autowired
     private TokenProvider jwtTokenUtil;
+
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/security/jwt/JWTAuthorizationFilter.java:52]: alignment of line 51 is wrong

**Review by dbmalkovsky — APPROVED** [review]
LGTM (with my limited knowledge in this area) after a few small cosmetic changes.

File: src/main/java/org/isf/security/jwt/JWTAuthorizationFilter.java [hunk 6]
```diff
@@ -42,10 +43,13 @@ public class JWTAuthorizationFilter extends BasicAuthenticationFilter {
 
 	private String HEADER_STRING = SecurityConstants.HEADER_STRING;
 	private String TOKEN_PREFIX = SecurityConstants.TOKEN_PREFIX;
-	private String SECRET = SecurityConstants.SECRET;
 	
+	@Autowired
+	private Environment env;
+
 	@Autowired
     private TokenProvider jwtTokenUtil;
```
Comment (mwithi) [inline src/main/java/org/isf/security/jwt/JWTAuthorizationFilter.java:51]:
```suggestion
        private TokenProvider jwtTokenUtil;
```

File: src/main/java/org/isf/security/jwt/JWTAuthorizationFilter.java (hunk 5 above)
Comment (mwithi) [reply src/main/java/org/isf/security/jwt/JWTAuthorizationFilter.java:52]: done, thanks!

**Comment by mwithi** [comment]
Superseded by https://github.com/informatici/openhospital-api/pull/209.

**Pull request closed without merging**